metals/src/main/scala/scala/meta/internal/metals/FormattingProvider.scala

```diff
@@ -215,11 +218,157 @@ final class FormattingProvider(
     } else Future.successful(false)
   }
 
+  private def highestDialect(): Option[ScalafmtDialect] =
+    buildTargets.all.toList
+      .map(_.fmtDialect)
+      .sorted
+      .lastOption
+
+  private def initialConfig(): String = {
+    val runnerDialect = highestDialect.filter(_ != ScalafmtDialect.Scala213)
+
+    val versionLine = s"""version = "${BuildInfo.scalafmtVersion}""""
+    val runnerLine = runnerDialect.map(dialect => s"runner.dialect = $dialect")
```

Comment:
I thought about that but it's hard to do absolutely correctly because of `globs`/`regex` - they have a free from.
Let's say there is the following config:
```
version = "2.7.5"
maxColumn = 100

fileOverride {
  'glob:$some-matcher' {
    maxColumn = 120
  }
}
```
In this case, we can't modify `fileOverride` as they might intersect each other.

Reply:
ach damn, so maybe in case of 3. we should just show a warning with a link to file-override configuration setting?

Reply:
Hmm. My first thought was that in most cases everything should be fine just to set global dialect but then I tried to do that for metals repo.
I got some non-significant rewrites related to newlines and 4 errors:
 - 3 of them about `inline`, `using`, `export` usage
 - `do while` syntax
 -  XML literals (will be fixed in the next scalafmt release?)
 
So, I agree that for the mixed project it would be better to use `fileOverride`.
The logic will be following:
- set the dialect using `fileOverride` per every non scala213 source directory if it `fileOverride` doesn't exist.
- if it exists - just make a warning. It will be hard to rewrite it because of issues with HOCON library +  I believe there aren't a lot of people who know about this setting.